.fixtures.yml

```diff
@@ -1,15 +1,21 @@
 ---
 fixtures:
   repositories:
-    inifile: 'https://github.com/puppetlabs/puppetlabs-inifile.git'
-    stdlib: 'https://github.com/puppetlabs/puppetlabs-stdlib.git'
-    postgresql: 'https://github.com/puppetlabs/puppet-postgresql.git'
-    firewall: 'https://github.com/puppetlabs/puppetlabs-firewall.git'
-    apt: 'https://github.com/puppetlabs/puppetlabs-apt.git'
-    concat: 'https://github.com/puppetlabs/puppetlabs-concat.git'
-    file_concat: 'https://github.com/electrical/puppet-lib-file_concat.git'
-    systemd: 'https://github.com/camptocamp/puppet-systemd.git'
-    cron: 'https://github.com/voxpupuli/puppet-cron.git'
+    provision: 'https://github.com/puppetlabs/provision.git'
+    puppet_agent: 'https://github.com/puppetlabs/puppetlabs-puppet_agent.git'
+    facts: 'https://github.com/puppetlabs/puppetlabs-facts.git'
     cron_core: 'https://github.com/puppetlabs/puppetlabs-cron_core.git'
     yumrepo_core: 'https://github.com/puppetlabs/puppetlabs-yumrepo_core.git'
     augeas_core: 'https://github.com/puppetlabs/puppetlabs-augeas_core.git'
+  forge_modules:
+    apt: 'puppetlabs/apt'
```

Comment:
for Vox Pupuli we always reference the git branches. That will pull in changes, for unit tests, before they are released. That allows us to detect breaking changes before they are released. For acceptance tests we pull in the latest forge releases.

Reply:
In principal I disagree. What is being tested should be what is being delivered.

Often I am dealing with the spaghetti of module dependencies in large control repo's. If I see a "tests passing" badge or CI, it's reasonable to expect you can deploy the module and know it's going to work with the dependencies stated in the metadata. Fixtures can't be the source of truth.

If a preview of upcoming changes is needed, that's easy enough to set up a CI job.

Reply:
Perfect example is this module. If I changed the fixtures to git, it would pull in the latest versions, and viola! all the acceptance tests pass. But alas, that's not reality.

Reply:
I really like that idea. Will come up with a PR for running tests with the HEAD of all deps.

There's also the (currently disabled) dependency checker, which is the dummy light of tests.

Reply:
> In principal I disagree. What is being tested should be what is being delivered.

that's what the acceptance tests do. The goal of the unit tests is to catch the errors before they are released (also pulling from github is usually faster compared to pulling from the forge).

But as mentioned in my initial comment, that's the strategy we (Vox Pupuli) use for Vox Pupuli. I think it makes sense, but I don't know what kind of strategy Puppet has.

Reply:
@bastelfreak learned something new today. updated commit.